--- scripts/setup-seafile-mysql.py      2013-10-19 13:33:12.000000000 +0200
+++ scripts/setup-seafile-mysql.py       2013-10-29 20:34:39.647473601 +0100
@@ -276,7 +276,7 @@
     def __init__(self):
         self.install_path = os.path.dirname(os.path.abspath(__file__))
         self.top_dir = os.path.dirname(self.install_path)
-        self.bin_dir = os.path.join(self.install_path, 'seafile', 'bin')
+        self.bin_dir = ''

     def check_pre_condiction(self):
         def error_if_not_exists(path):
