--- sope-gdl1/MySQL/MySQL4Channel.m	2009-11-28 00:43:27.731467911 +0100
+++ sope-gdl1/MySQL/MySQL4Channel.m.new	2009-11-28 00:42:10.683968806 +0100
@@ -634,11 +634,11 @@
 
     if ((merrno = mysql_errno(self->_connection)) != 0) {
       const char *error;
+      error = mysql_error(self->_connection);
       
       if (isDebuggingEnabled)
         NSLog(@"%@   cannot use result: '%s'", self, error);
       
-      error = mysql_error(self->_connection);
       return [MySQL4Exception exceptionWithName:@"FetchFailed" 
                               reason:[NSString stringWithUTF8String:error]
                               userInfo:nil];
@@ -763,12 +763,16 @@
 				    @"  c_lastmodified INT NOT NULL,\n"
 				    @"  c_version INT NOT NULL,\n"
 				    @"  c_deleted INT NULL\n"
-				    @")");
+				    @")\n"
+				    @" DEFAULT CHARACTER SET latin1\n"
+				    @" DEFAULT COLLATE latin1_general_ci");
 static NSString *sqlFolderACLFormat = (@"CREATE TABLE %@ (\n"	\
 				       @"  c_uid VARCHAR (256) NOT NULL,\n"
 				       @"  c_object VARCHAR (256) NOT NULL,\n"
 				       @"  c_role VARCHAR (80) NOT NULL\n"
-				       @")");
+				       @")\n"
+				       @" DEFAULT CHARACTER SET latin1\n"
+				       @" DEFAULT COLLATE latin1_general_ci");
 
 - (NSException *) createGCSFolderTableWithName: (NSString *) tableName
 {
