--- sope-appserver/SoOFS/GNUmakefile.preamble.org	2009-11-24 15:05:35.971466918 +0100
+++ sope-appserver/SoOFS/GNUmakefile.preamble	2009-11-24 15:04:47.001467300 +0100
@@ -3,6 +3,7 @@
 # TODO: need to fix COMPILE_FOR_GSTEP_MAKE?
 ADDITIONAL_CPPFLAGS  += -Wall -DCOMPILE_FOR_GSTEP_MAKE=1
 ADDITIONAL_OBJCFLAGS += -Wall -Wno-import -Wno-protocol
+ADDITIONAL_LDFLAGS += "-Wl,--no-as-needed"
 
 SOPE_ROOT=../..
 CORE_ROOT=$(SOPE_ROOT)/sope-core
@@ -83,6 +83,7 @@
 ifneq ($(findstring darwin, $(GNUSTEP_TARGET_OS)), darwin)
 # and neither does MacOSX? ...
-sope_TOOL_LIBS += -lcrypt
+$(SOPED_NAME)_TOOL_LIBS += -lcrypt
+libSoOFS_LIBRARIES_DEPEND_UPON += -lcrypt
 endif
 endif
 
