--- sope-core/NGExtensions/NGLogging.subproj/NGLogSyslogAppender.m	2009-11-28 00:55:21.091466955 +0100
+++ sope-core/NGExtensions/NGLogging.subproj/NGLogSyslogAppender.m.new	2009-11-28 00:54:15.053967324 +0100
@@ -86,7 +86,7 @@
   
   formattedMsg = [self formattedEvent:_event];
   level = [self syslogLevelForLogLevel:[_event level]];
-  syslog(level, [formattedMsg cString]);
+  syslog(level, "%s", [formattedMsg cString]);
 }
 
 - (int)syslogLevelForLogLevel:(NGLogLevel)_level {
